plugins/woocommerce/src/Blocks/Patterns/PTKClient.php

```diff
@@ -14,6 +14,67 @@ class PTKClient {
 	 */
 	const PATTERNS_TOOLKIT_URL = 'https://public-api.wordpress.com/rest/v1/ptk/patterns/';
 
+	/**
+	 * The schema for the patterns toolkit.
+	 *
+	 * @var array
+	 */
+	private $schema;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->schema = [
+			'type'  => 'array',
+			'items' => [
+				'type'       => 'object',
+				'properties' => [
+					'ID'         => [
+						'type'     => 'integer',
+						'required' => true,
+					],
+					'site_id'    => [
+						'type'     => 'integer',
+						'required' => true,
+					],
+					'title'      => [
+						'type'     => 'string',
+						'required' => true,
+					],
+					'name'       => [
+						'type'     => 'string',
+						'required' => true,
+					],
+					'html'       => [
+						'type'     => 'string',
+						'required' => true,
+					],
+					'categories' => [
+						'type'                 => 'object',
+						'additionalProperties' => [
+							'type'       => 'object',
+							'properties' => [
+								'slug'        => [
+									'type'     => 'string',
+									'required' => true,
+								],
+								'title'       => [
+									'type'     => 'string',
+									'required' => true,
+								],
```

Comment:
Should we enforce a [`minLength`](https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/#strings) for `slug` and `title` to make sure they are not empty strings?

Reply:
I don't think that we should do it. The important thing is that the type is a string.